velox/exec/HashProbe.cpp

```diff
@@ -589,8 +607,8 @@ int32_t HashProbe::evalFilter(int32_t numRows) {
 }
 
 void HashProbe::ensureLoadedIfNotAtEnd(column_index_t channel) {
-  if (core::isSemiJoin(joinType_) || core::isAntiJoin(joinType_) ||
-      results_.atEnd()) {
+  if (core::isRightSemiJoin(joinType_) || core::isSemiJoin(joinType_) ||
```

Comment:
Is this needed for right semi join. I think it may not be needed as we do not return any of the probe-side rows.